ocaml/xapi-aux/networking_info.ml

```diff
@@ -0,0 +1,79 @@
+(*
+ * Copyright (C) 2006-2009 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+module Net = Network_client.Client
+
+let get_hostname () = try Unix.gethostname () with _ -> ""
+
+exception Unexpected_address_type of string
+
+(* Try to get all FQDNs, avoid localhost *)
+let dns_names () =
+  let hostname = get_hostname () in
+  let fqdns =
+    Unix.getaddrinfo hostname "" [Unix.AI_CANONNAME]
+    |> List.map (fun x -> x.Unix.ai_canonname)
+  in
+  hostname :: fqdns
+  |> List.filter_map (fun x ->
+         let x = Astring.String.trim x in
+         if
+           String.equal "" x
+           || String.equal "localhost" x
+           || Ipaddr.of_string x |> Stdlib.Result.is_ok
+         then
+           None
+         else
+           Some x)
+  |> Astring.String.uniquify
+
+let ip_addr_of_string ip =
+  Ipaddr.of_string ip
+  |> Stdlib.Result.to_option
+  |> Option.map (function
+       | Ipaddr.V4 addr ->
+           Cstruct.of_string (Ipaddr.V4.to_octets addr)
+       | Ipaddr.V6 addr ->
+           Cstruct.of_string (Ipaddr.V6.to_octets addr))
+
+let get_management_ip_addr ~dbg =
```

Comment:
Small note on the name: `_addr` is redundant, isn't it? What else would we expect behind the name?

Reply:
This is the historical name of the function. This also reflects the name commonly used in Unix: `ip_addr`